assisted-iso-create/main.go

```diff
@@ -79,31 +27,21 @@ func main() {
 		log.Fatal(err.Error())
 	}
 
-	ignitionFilePath, err := setIgnitionConfigToFile(Options.WorkDir, Options.IgnitionConfig, log)
-	if err != nil {
-		log.Fatal(err.Error())
-	}
+	var s3Client *s3wrapper.S3Client
+	log.Infof("S3 parameters: bucket %s, region %s", Options.S3Config.S3Bucket, Options.S3Config.Region)
 
-	createImageFile, err := embedIgnitionIntoISO(Options.WorkDir, ignitionFilePath, Options.ImageName, Options.BaseISOFile, log)
+	s3Client = s3wrapper.NewS3Client(&Options.S3Config, log)
+	if s3Client == nil {
+		log.Fatal("failed to create S3 client, ", err)
+	}
+	err = s3Client.UploadFile(ctx, Options.BaseISOFile, Options.ImageName)
 	if err != nil {
-		log.Fatal(err.Error())
+		log.Fatalf("Failed to upload file %s as object %s", Options.BaseISOFile, Options.ImageName)
 	}
-
-	if Options.UseS3 {
-		log.Infof("S3 parameters: bucket %s, region %s", Options.S3Config.S3Bucket, Options.S3Config.Region)
-
-		s3Client := s3wrapper.NewS3Client(&Options.S3Config, log)
-		if s3Client == nil {
-			log.Fatal("failed to create S3 client, ", err)
-		}
-
-		err = uploadCreatedISOToS3(s3Client, createImageFile, Options.ImageName, log)
-		if err != nil {
-			log.Fatal(err.Error())
-		}
-	} else {
-		log.Println("Using local storage for image")
+	_, err = s3Client.UpdateObjectTimestamp(ctx, Options.ImageName)
```

Comment:
When uploading a base image should we overwrite the existing one or should we create a new one?